src/modules/emote_menu/twitch/EmoteMenu.jsx

```diff
@@ -58,29 +59,46 @@ export default class EmoteMenuModule {
     }
 
     const legacyContainer = document.querySelector(LEGACY_BTTV_EMOTE_PICKER_BUTTON_CONTAINER_SELECTOR);
-    const emoteMenuEnabled = settings.get(SettingIds.EMOTE_MENU);
+    const emoteMenuValue = settings.get(SettingIds.EMOTE_MENU);
 
-    if (emoteMenuEnabled && legacyContainer == null) {
-      const container = document.querySelector(CHAT_SETTINGS_BUTTON_CONTAINER_SELECTOR);
-      if (container == null) {
-        return;
-      }
-      const rightContainer = container.lastChild;
+    if (emoteMenuValue !== EmoteMenuTypes.NONE) {
       const buttonContainer = document.createElement('div');
```

Comment:
this new logic will create a button container even if the button is already rendered. we should keep the prior optimizations in tact

Reply:
Prior logic doesn't take into account the placement I believe, so even if the button is already rendered we should try replace its location on the DOM? I may be wrong however?

Reply:
could we check for either button?

Reply:
this doesn't appear resolved. to fix this, we could split the two button references up between `legacyButtonContainer` and `buttonContainer`, which would allow you to check existence of either when toggling between them.